src/arrayMethodJoin.js

```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let separator2 = separator;
+
+    if (typeof separator === 'undefined') {
+      separator2 = ',';
+    }
+
+    let testString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (typeof this[i] === 'object' || typeof this[i] === 'undefined') {
+        testString += `${separator2}`;
```

Comment:
```suggestion
        testString += separator;
```
You can just add it without interpolation